### ISSUE TITLE
With or without single quotes?

### ISSUE THREAD

**Issue by krilbe** [dotnet/efcore#17554]
The docs/code comments do not specify if the returned literal (always) includes cingle quotes. From a bug I encounter in a dependent package, it would seem that if the literal includes a character encoding prefix, the single quotes are included. Good. But are they also included if no encoding prefix is present? I would hope so, but it would be better if the docs were specific about this.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 10657614-766e-bd5a-838e-2f8cd53f527e
* Version Independent ID: 44e27f77-aa67-2730-0b9e-cdcebe80919e
* Content: [RelationalTypeMapping.GenerateSqlLiteral(Object) Method (Microsoft.EntityFrameworkCore.Storage)](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.storage.relationaltypemapping.generatesqlliteral?view=efcore-2.1)
* Content Source: [dotnet/xml/Microsoft.EntityFrameworkCore.Storage/RelationalTypeMapping.xml](https://github.com/aspnet/EntityFramework.ApiDocs/blob/live/dotnet/xml/Microsoft.EntityFrameworkCore.Storage/RelationalTypeMapping.xml)
* Product: **ef-core-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **divega**

**Comment by ajcvickers** [comment]
@krilbe It generates a literal value that is safe to use, the details of which vary between type and database provider. If you're seeing a situation where this isn't the case, then please post a small, runnable project/solution or complete code listing that demonstrates this behavior.

**Comment by krilbe** [comment]
It's been a while but I think the bug I encountered was that the dependent package put single quotes around the returned string when the returned string includes a character encoding prefix. I think the intention was to add single quotes when the returned value wasn't already quoted, and failed to take the possibility of an encoding prefix into consideration. I think I got it working by simply assuming that this method always will return a complete string literal with single quotes even if no character encoding prefix is present, and simply remove the dependent package's code to add quotes.
